src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,126 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import cn from 'classnames';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo,
+  onHide: () => void,
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, onHide }) => {
+  const {
+    id,
+    userId,
+    title,
+    completed,
+  } = todo;
+
+  const [user, setUser] = useState<User>();
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const getUserFromApi = async () => {
+    try {
+      const userFromApi = await getUser(userId);
+
+      setUser(userFromApi);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getUserFromApi();
+  }, []);
+
+  const displayModal = !isLoading && !isError && user;
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
+      {isLoading && (
+        <Loader />)}
+      {displayModal && (
+        <div
+          className={cn(
+            'modal-card',
+            'notification',
+            {
+              'is-success': completed,
+              'is-danger': !completed,
+            },
+          )}
+        >
           <header className="modal-card-head">
             <div
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${id}`}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
               type="button"
               className="delete"
               data-cy="modal-close"
+              onClick={onHide}
             />
           </header>
 
           <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+            <p className="block has-text-dark" data-cy="modal-title">
+              {title}
             </p>
 
             <p className="block" data-cy="modal-user">
               {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+              <strong
+                className={cn({
+                  'has-text-danger': !completed,
+                  'has-text-success': completed,
+                })}
+              >
+                {completed ? 'Done' : 'Planned'}
+              </strong>
 
               {' by '}
 
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
+              <a href={`mailto:${user?.email}`}>
+                {user?.name}
               </a>
```

Comment:
I can't see author
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/84296041/865d549e-ce8e-4a5d-8b0e-f23b6266cd4f)